atc/exec/step_metadata.go

```diff
@@ -62,5 +64,15 @@ func (metadata StepMetadata) Env() []string {
 		env = append(env, "ATC_EXTERNAL_URL="+metadata.ExternalURL)
 	}
 
+	if metadata.CreatedBy != nil {
+		if metadata.CreatedBy.UserID != "" {
```

Comment:
I don't know that we can be sure of this order of precedence - `UserID` with the GitHub connector returns the user's numeric ID which isn't super useful - seems like the login or email would be more useful there.

What do you think of just JSON encoding the claims and sending them along? Seems like UserID might be useful in some contexts and UserName useful in others, so maybe it should be up to the resource type to choose.